Counterexamples/MapFloor.lean

```diff
@@ -49,8 +49,7 @@ open scoped Polynomial
 
 /-- The integers with infinitesimals adjoined. -/
 def IntWithEpsilon :=
-  ℤ[X]
-deriving CommRing, Nontrivial, Inhabited
+  ℤ[X] deriving Nontrivial
```

Comment:
Did these `deriving` instances not work? If not, it would be good to understand why.

Reply:
They did not work (the other two).

Reply:
[Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60deriving.60.20not.20working)